### PR TITLE
[ricos-editor] feat: rich content schema public api

### PR DESCRIPTION
**Pull request by maimonav**
Rich Content schema goes public! (old news, but now in the Editor).
By applying `schemaType: 'RichContent'`, it changes the schema type of the Editor, making every Editor public API with the new `RichContent` schema.

 - Schema entity, toggles between `Draft` & `RichContent` schemas.
 - Type BREAKING, from `DraftContent` to `DraftContent | RichContent`.
 - A step towards removing Draft completely from our code.

This will allow users do migrate `DraftContent` into `RichContent` and removing usages of Draft converters (Soon Bye Bye Draft).

### CODE REVIEW

File: packages/ricos-editor/web/src/tiptap/RicosEditor.tsx [hunk 1]
```diff
@@ -139,16 +145,14 @@ class RicosEditor extends React.Component<Props, RicosEditorState> implements Ri
       this.props.editor.publishFirstEdit();
     }
     this.isLastChangeEdit = true;
+    console.log(content);
```
Comment (maimonav) [inline packages/ricos-editor/web/src/tiptap/RicosEditor.tsx:148]: remove

File: packages/ricos-orchestrator/web/src/ricos-schema.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+}
+
+export class DraftSchema {
+  editor!: RicosEditor;
+
+  experiments: RicosEditorProps['experiments'];
+
+  constructor(experiments: RicosEditorProps['experiments']) {
+    this.experiments = experiments;
+  }
```
Comment (oded-soffrin) [inline packages/ricos-orchestrator/web/src/ricos-schema.ts:123]: AKA:

```suggestion
  constructor(private experiments: RicosEditorProps['experiments']) {
  }
```

Comment (oded-soffrin) [inline packages/ricos-orchestrator/web/src/ricos-schema.ts:123]: AKA:

```suggestion
  constructor(private experiments: RicosEditorProps['experiments']) {
  }
```

Comment (maimonav) [reply packages/ricos-orchestrator/web/src/ricos-schema.ts]: Thanks 🙏 
I prefer it the "Old Fashion" way :).

File: packages/ricos-common/web/src/RicosEngine.tsx [hunk 3]
```diff
@@ -44,6 +44,13 @@ export class RicosEngine extends Component<EngineProps> {
       experiments,
     } = this.props;
 
+    if (
+      this.props.schemaType === 'RichContent' ||
+      !!(this.props.content as RichContent | undefined)?.nodes
+    ) {
+      throw new Error('Draft editor supports only Draft content schema.');
+    }
+    const content: DraftContent | undefined = this.props.content as DraftContent | undefined;
```
Comment (noamtwix) [inline packages/ricos-common/web/src/RicosEngine.tsx:53]:
```suggestion
    const content = this.props.content as DraftContent | undefined;
```

File: packages/ricos-editor/web/src/tiptap/RicosEditor.tsx [hunk 4]
```diff
@@ -65,11 +55,22 @@ class RicosEditor extends React.Component<Props, RicosEditorState> implements Ri
...
+              (newProps.injectedContent as RichContent).nodes
+            ) &&
+            !isEqual(
+              (this.props.injectedContent as RichContent).documentStyle,
+              (newProps.injectedContent as RichContent).documentStyle
+            )
+          : !isEqual(this.props.injectedContent, newProps.injectedContent)
+      ) {
+        this.props.schema?.setContent(newProps.injectedContent);
+      }
```
Comment (noamtwix) [inline packages/ricos-editor/web/src/tiptap/RicosEditor.tsx:73]: Suggestion:
Maybe the next person who comes across this `if` statement doesn't waste time understanding.

And another thing, are you sure is `&&` and not `||`?

```suggestion
    if (newProps.injectedContent) {
      const isRichContentSchema = newProps.schemaType === 'RichContent';
      const isRichContentSchemaContentChanged = !isEqual(
              (this.props.injectedContent as RichContent).nodes,
              (newProps.injectedContent as RichContent).nodes;
      const isDocumentStyleChanged = !isEqual(
              (this.props.injectedContent as RichContent).documentStyle,
              (newProps.injectedContent as RichContent).documentStyle;
      const isDraftSchemaContentChanged =   !isEqual(this.props.injectedContent, newProps.injectedContent);
      if (
        isRichContentSchema
          ? isRichContentSchemaContentChanged && isDocumentStyleChanged
          : isDraftSchemaContentChanged
      ) {
        this.props.schema?.setContent(newProps.injectedContent);
      }
```

File: packages/ricos-orchestrator/web/src/ricos-schema.ts [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    }
+    if (initialContent && !!(initialContent as RichContent).nodes && schemaType === 'Draft') {
+      throw new Error(
+        'Mismatched schema: "content" of type RichContent can not be used with "schemaType" Draft.'
+      );
+    }
+
+    this.initialTiptapContent = draftToTiptap(
+      ensureDraftContent(initialContent || getEmptyDraftContent())
+    );
```
Comment (noamtwix) [inline packages/ricos-orchestrator/web/src/ricos-schema.ts:42]: What is the reason to transfer RichContent to DraftContent and then to TiptapContent?
Don't we have some `richContentToTiptap` function (equivalent to `draftToTiptap`)?

File: packages/ricos-editor/web/src/tiptap/RicosEditor.tsx (hunk 4 above)
Comment (maimonav) [reply packages/ricos-editor/web/src/tiptap/RicosEditor.tsx:73]: @noamtwix 

> Suggestion:
> Maybe the next person who comes across this if statement doesn't waste time understanding.
Thanks, I'll refactor it :)

> And another thing, are you sure is && and not ||?
Great catch! thank you 🔥

Comment (noamtwix) [reply packages/ricos-editor/web/src/tiptap/RicosEditor.tsx:73]: @maimonav 

No problem.
When it's easy to read it's easy to catch ;)

File: packages/ricos-common/web/src/RicosEngine.tsx (hunk 3 above)
Comment (maimonav) [reply packages/ricos-common/web/src/RicosEngine.tsx:53]: Thanks :)

**Pull request merged**